src/User.php

```diff
@@ -6592,7 +6600,8 @@ public function applyGroupsRules()
 
             $data = [
                 'groups_id' => $group_id,
-                'users_id'  => $this->getId()
+                'users_id'  => $this->getId(),
+                'is_dynamic' => 1
```

Comment:
There is a unicity key on `groups_id+users_id` for the `glpi_groups_users` table. If you add the `is_dynamic=1` criteria, then if the relation has already been added manually (`is_dynamic=0`), you will try to create the same one  with `is_dynamic=1` and it will result in a SQL error.

IMHO, if the relation already exists, dynamic or not, you should not change it.